src/generic-provider-views/index.js

```diff
@@ -253,6 +254,27 @@ module.exports = class View {
     })
   }
 
+  handleAuth () {
+    let urlId = Math.floor(Math.random() * 999999) + 1
+    let redirect = `${location.href}${location.search ? '&' : '?'}id=${urlId}`
+
+    const authState = btoa(JSON.stringify({ redirect }))
+    const link = `${this.plugin.opts.host}/connect/${this.Provider.authProvider}?state=${authState}`
+
+    let authWindow = window.open(link, '_blank')
+
+    let checkAuth = () => {
```

Comment:
On the related note, I was thinking about using `function` keywords for functions (back to the roots), to make a clear distinction from variables. What do you think? Technically `const = () =>` is correct, new and shiny, and might work well for one-liners and in those cases when you need to access `this` of a parent, but. Just a thought here, I’m in doubt myself.